scripts/policy/frameworks/management/agent/main.zeek

```diff
@@ -1049,6 +1054,14 @@ event Broker::peer_added(peer: Broker::EndpointInfo, msg: string)
 	local sni = supervisor_network_info();
 
 	if ( peer$network$address == sni$address && peer$network$bound_port == sni$bound_port )
+		g_supervisor_peered = T;
+
+	# If the Supervisor hasn't yet peered with us, don't broadcast
+	# notify_agent_hello. Doing so would exposes a race: we might receive
+	# commands from the controller that lead to requests to the Supervisor
+	# that it won't yet receive. It's easier to handle this here than to
+	# push the wait into all types of received commands.
+	if ( g_supervisor_peered == F )
 		return;
```

Comment:
I don't have overall flow well in my head, but just looking at this change in isolation I'm wondering about the change in behavior if I'm reading the diff right. Previously it would return immediately if it's the supervisor; now it always proceeds to rest of the handler in that case. That intentional?